pylint/checkers/stdlib.py

```diff
@@ -287,6 +317,42 @@ def _check_open_mode(self, node):
                 self.add_message('bad-open-mode', node=node,
                                  args=mode_arg.value)
 
+    def _check_invalid_envvar_value(self, call_arg, infer):
+        if isinstance(call_arg, Call):
+            call_arg = call_arg.inferred()
+
+            if not call_arg:
+                return
+
+            call_arg = call_arg[0]
+
+        if call_arg is astroid.Uninferable:
+            return
+
+        if isinstance(call_arg, Const):
+            if not isinstance(call_arg.value, (six.string_types, six.text_type)):
```

Comment:
`six.string_types` should be enough since it will contain both `unicode` and `str` on Python 2.